### ISSUE TITLE
ct-clustertool - errored connection

### ISSUE THREAD

**Issue by AndrewGorton** [google/certificate-transparency#1081]
I'm trying to start up the xjson-server in non-standalone mode.

I have a 3-way cluster of CoreOS (Stable) servers in an etcd2 cluster. I have created the etcd2 entries with:-

```
# etcdctl mkdir /root
# etcdctl mkdir /root/entries
# etcdctl mkdir /root/nodes
# etcdctl set /root/sequence_mapping ""
# etcdctl set /root/serving_sth ""
# etcdctl set /root/cluster_config ""
```
I start the ct-clustertool with:-

```
# ct-clustertool initlog --key=ct.key --etcd_servers=ip-xyz-1.eu-west-1.compute.internal:4001,ip-xyz-2.eu-west-1.compute.internal:4001,ip-xyz-3.eu-west-1.compute.internal:4001
I1217 14:46:46.087555    18 thread_pool.cc:146] ThreadPool starting with 1 threads
W1217 14:46:46.094620    18 connection_pool.cc:257] Setting client minimum TLS protocol to tlsv12
I1217 14:46:46.095430    18 connection_pool.cc:289] Loading openssl default trusted root certificates...
I1217 14:46:46.189077    20 etcd.cc:1034] Etcd version: {"etcdserver":"2.2.0","etcdcluster":"2.2.0"}
I1217 14:46:46.253551    19 masterelection.cc:546] /election/clustertool: Became master
I1217 14:46:46.253721    18 thread_pool.cc:146] ThreadPool starting with 4 threads
I1217 14:46:46.352272    18 sqlite_db.cc:92] New SQLite database created in /tmp/clustertooldb
W1217 14:46:46.352371    18 sqlite_db.cc:143] SQLite "synchronous" pragma set to FULL
W1217 14:46:46.352385    18 sqlite_db.cc:146] SQLite running with batched transactions, you should set sqlite_synchronous_mode = FULL !
W1217 14:46:46.376621    18 clustertool_main.cc:112] Using default ClusterConfig
I1217 14:46:46.380018    18 clustertool_main.cc:126] Using config:
minimum_serving_nodes: 2
minimum_serving_fraction: 0.75
I1217 14:46:46.409433    18 clustertool_main.cc:179] OK
W1217 14:47:46.400647    19 connection_pool.cc:335] Releasing errored connection to ip-xyz-1.eu-west-1.compute.internal:4001
W1217 14:47:47.349406    19 connection_pool.cc:360] error flag (0x41): READING TIMEOUT : Operation now in progress
W1217 14:47:47.468257    19 connection_pool.cc:335] Releasing errored connection to ip-xyz-1.eu-west-1.compute.internal:4001
W1217 14:47:47.468292    19 connection_pool.cc:360] error flag (0x41): READING TIMEOUT : Operation now in progress
W1217 14:47:47.468590    19 connection_pool.cc:335] Releasing errored connection to ip-xyz-1.eu-west-1.compute.internal:4001
W1217 14:47:47.468621    19 connection_pool.cc:360] error flag (0x41): READING TIMEOUT : Operation now in progress
```

I check reachability with

```
# curl ip-xyz-1.compute.internal:4001/v2/keys
{"action":"get","node":{"dir":true,"nodes":[{"key":"/root","dir":true,"modifiedIndex":9162,"createdIndex":9162},{"key":"/election","dir":true,"modifiedIndex":9199,"createdIndex":9199},{"key":"/coreos.com","dir":true,"modifiedIndex":6,"createdIndex":6}]}}
```
Has the above worked, or am I missing something?

Thanks,

Andrew.

**Comment by AlCutter** [comment]
Bit of a red herring here, I think.

The line:
```
I1217 14:46:46.409433    18 clustertool_main.cc:179] OK
```
means the `initlog` command has worked.
ISTR there's an occasional situation with libevent which causes the eventloop to hang around for ~50s before finally exiting.

The `TIMEOUT` stuff is, I believe, coming from the etcd client watches (which are just hanging `GET`s).  These `TIMEOUT`s messages should probably be squelched in that particular case.

TL/DR; your log should be initialised ok, ignore the scary messages.

**Comment by AndrewGorton** [comment]
Much obliged - thanks!
